### PR TITLE
adjusted format of logs

### PR DESCRIPTION
**Pull request by LBolte29**
# Description

adjusted format of logs

# How has this been tested?

checked logs

### CODE REVIEW

**Comment by DidierRLopes** [comment]
Comparing the previous logs:
```
I|sha:3f766c95|2022-01-26T18:11:25+0000|INFO|gamestonk_terminal.loggers|setup_logging|183|Logging configuration finished|none|
I|sha:3f766c95|2022-01-26T18:11:25+0000|INFO|gamestonk_terminal.loggers|setup_logging|184|Logging set to file|none|
I|sha:3f766c95|2022-01-26T18:11:25+0000|INFO|gamestonk_terminal.loggers|setup_logging|185|Verbosity set to 20|none|
I|sha:3f766c95|2022-01-26T18:11:25+0000|INFO|__main__|terminal|219|Terminal started|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.disc_controller|call_gainers|0|START params: -l 4|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_gainers|0|START params: num_stocks=4, export=''|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_model|get_gainers|0|START|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_model|get_gainers|0|END|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_gainers|0|END|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.stocks.discovery.disc_controller|call_gainers|0|END|none|
I|sha:3f766c95|2022-01-26T18:11:26+0000|INFO|gamestonk_terminal.terminal_helper|print_goodbye|315|Terminal stopped|none|
```

to these ones:
```
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.loggers|setup_logging|222|Logging configuration finished||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.loggers|setup_logging|223|Logging set to file||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.loggers|setup_logging|224|Verbosity set to 20||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|__main__|terminal|220|START||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<__main__.TerminalController object at 0x7fe3c1bef0d0>;'/stocks/disc/gainers/losers/exit'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'home/stocks/disc/gainers/losers/exit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<__main__.TerminalController object at 0x7fe3c1bef0d0>;'home'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|call_home|0|START|<__main__.TerminalController object at 0x7fe3c1bef0d0>;[]|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|call_home|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'stocks/disc/gainers/losers/exit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<__main__.TerminalController object at 0x7fe3c1bef0d0>;'stocks'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<gamestonk_terminal.stocks.stocks_controller.StocksController object at 0x7fe3f0e4b7c0>;'disc'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<gamestonk_terminal.stocks.discovery.disc_controller.DiscoveryController object at 0x7fe3c21c2f10>;'gainers'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:27+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_gainers|0|START||{'num_stocks': 5, 'export': ''}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_gainers|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'losers/exit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<gamestonk_terminal.stocks.discovery.disc_controller.DiscoveryController object at 0x7fe3c21c2f10>;'losers'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_losers|0|START||{'num_stocks': 5, 'export': ''}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.stocks.discovery.yahoofinance_view|display_losers|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'exit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|START|<gamestonk_terminal.stocks.discovery.disc_controller.DiscoveryController object at 0x7fe3c21c2f10>;'exit'|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|call_exit|0|START|<gamestonk_terminal.stocks.discovery.disc_controller.DiscoveryController object at 0x7fe3c21c2f10>;[]|||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|call_exit|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'quit/quit/quit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'quit/quit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|168|||{'remaining queue': 'quit'}||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.parent_classes|switch|0|END||||
I|sha:bbaca4f0|179588f6-2bd2-46fc-86ce-437dec1987b8|1643902047|2022-02-03T15:27:28+0000|INFO|gamestonk_terminal.terminal_helper|print_goodbye|315|END||||
```

Some notes:
* gamestonk_terminal.parent_classes shouldn't appear. We should display the controller that is inheriting from this file
* An address such as `<__main__.TerminalController object at 0x7fe3c1bef0d0>` shouldn't appear
* I don't think we want the `{ }` within the logs. I would even avoid the `'` character. I feel like less "tricky" characters will make the data processing easier in the long run
* Can you in this PR describe the change of format on the loggers? And can we run it by @Chavithra and @aia ?
   *  I feel like the start time is very much redundant given that if I understood correctly this corresponds to the int(next field)
   * Can we confirm with @aia that he agrees with the session ID also being present. 
   * I think it would be good to have a design document filled in so everyone is happy with this.

File: gamestonk_terminal/loggers.py [hunk 1]
```diff
@@ -146,6 +176,9 @@ def get_commit_hash() -> None:
 def setup_logging() -> None:
     """Setup Logging"""
 
+    START_TIME = int(time.time())
+    LOGGING_ID = cfg.LOGGING_ID if cfg.LOGGING_ID else ""
```
Comment (DidierRLopes) [inline gamestonk_terminal/loggers.py:180]: This code seems redundant to me, but may be missing something

**Review by DidierRLopes — CHANGES_REQUESTED** [review]

**Comment by LBolte29** [comment]
Thanks for the review @DidierRLopes :) 
1. The reason the parent_classes appears in the logger is that the logger actually is implemented there. Check https://github.com/GamestonkTerminal/GamestonkTerminal/pull/1270/
Can ofc change this ;)
2. Agree on the address, thought I'd tackle that in the following iteration and just get out the PR. Will do :) 
3. Why do you feel that data processing will be easier in the long run? What is your suggestion to store this? I guess a smart way to store this info depends on what framework we will use to store and process the data. If we want to distribute the load then we should think about that. Happy to discuss this further ;)
4. Will work on a design document, that we can discuss together with @aia - This is needed for sure. Surely this will also need a couple of iterations ;) 
About the session ID: The start time would be the session ID indeed. If we want to use the information of a session in the analytics, then we have 2 options:
a) we need to put it in when we write it or when we process it (taking it from the title). If we do the latter that means, that we can only have 1file per session, which (in case we distributed processing) we will increase the likelihood of hitting/ the magnitude of the problems of skewed partitioning / small files problem.

**Comment by DidierRLopes** [comment]
> 1. Added logger to baseclass controller #1270


1. Yes, having parent classes logger is just not useful IMO. I think what we would in case user does something like `python3 terminal.py stocks/disc/gainers` is: Terminal started, stocks controller, disc controller, gainers function is called, display_gainers on _view, gainers on _model and then the path back. 
2. I'd suggest for us to make this PR as draft and merge whenever we are happy with logging format so we can iterate on this PR and come back here in the future for reference, rather than looking at multiple ones.
3. I just feel like is adding extra complexity to the logs, i.e. what are we gaining out of having `{'num_stocks': 5, 'export': ''}` instead of `num_stocks:5, export:''`? Seems like the first will need the same data processing as the latter and more, unless you are planning to just read this as a dict? Another thing to think is that we want to logs to be relatively easy to read so we can quickly understand what's going on.
4. Yea, I understand that. And forget what I said about start time, I see what you did there. My gut feeling tells me that we don't need this, but then on the other hand I see how this would simplify the log receiver pipeline. So am happy either way. Let's decide on design so we can proceed with this :)

**Comment by LBolte29** [comment]
Thx for the quick answer :) 
1. Sure, Maybe in the case of the switch, we could just replace the parent_class with the first argument (which actually is the particular controller)? What do you think?
2. Sure
3. If we use spark, then we should leverage some internal optimizations to read complex fields. If we just have it as a list, this would imply string processing of each line, which may be a bottleneck. I was planning to read this as a dict, ye. But (see following)
3 + 4. But in any case, I agree we should have a design doc, where we justify these choices and then can make informed decisions together weighing out different aspects...
I will draft a first one this afternoon :)

**Comment by LBolte29** [comment]
https://openbb.atlassian.net/wiki/spaces/PROD/pages/64094309/Terminal+Logging+Format

**Comment by LBolte29** [comment]
Once we settled on a format and have settled on a version of the PR we can agree with, we should probably try to read the logs with the data processing framework we agree on. This will validate that the framework can handle the logs OOTB!
Tagged you on a doc yesterday btw
Very excited about this! :)

**Comment by DidierRLopes** [comment]
> Once we settled on a format and have settled on a version of the PR we can agree with, we should probably try to read the logs with the data processing framework we agree on. This will validate that the framework can handle the logs OOTB!
> Tagged you on a doc yesterday btw
> Very excited about this! :)

Not sure we will have time for that, but that would be great indeed! The idea is to agree on the format and push it everywhere.

**Review by Chavithra — APPROVED** [review]
Looks good to me


Few questions :
1/ Isn't column 1 with content like this : `I`, `X`, `D`
Is a duplicate with the column displaying : `INFO`, `ERROR`, `DEBUG`

2/ How do you plan to extract messages like this :
- `Running Python 3.8.7 on Windows`
Maybe we need some documented keyword to be able to easily extract those :
- `PYTHON_VERSION::Running Python 3.8.7 on Windows`

**Review by DidierRLopes — APPROVED** [review]
🚀  🌔

**Pull request merged**